test/unit/ObjectMakrTests/make.js

```diff
@@ -0,0 +1,38 @@
+define(["mocks"], function (mocks) {
+    return function () {
+        var expect = require("chai").expect;
+
+        it("object's name is John the Penguin", function () {
+            // Arrange
+            var objectMaker = mocks.mockObjectMakr();
+
+            // Act
+            var madeObject = objectMaker.make(mocks.mockClassExample, {name: "John", weight: 213, age: 4});
+
+            // Assert
+            expect(madeObject["name"]).to.equal("John the Penguin");
```

Comment:
Harping on [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns), just reading this test I have no idea why the name would end with `"the Penguin"`. This test is essentially testing two things: that a single property is copied, and that `onMake` works.